From j-nomura@ce.jp.nec.com Wed Mar 15 05:27:20 2006
Message-ID: <44181697.2070709@ce.jp.nec.com>
Date: Wed, 15 Mar 2006 08:28:55 -0500
From: "Jun'ichi Nomura" <j-nomura@ce.jp.nec.com>
To: Greg KH <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org
Subject: kobject: fix build error if CONFIG_SYSFS=n


Moving uevent_seqnum and uevent_helper to kobject_uevent.c
because they are used even if CONFIG_SYSFS=n
while kernel/ksysfs.c is built only if CONFIG_SYSFS=y,

Signed-off-by: Jun'ichi Nomura <j-nomura@ce.jp.nec.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 kernel/ksysfs.c      |    3 ---
 lib/kobject_uevent.c |    2 ++
 2 files changed, 2 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/kernel/ksysfs.c
+++ gregkh-2.6/kernel/ksysfs.c
@@ -15,9 +15,6 @@
 #include <linux/module.h>
 #include <linux/init.h>
 
-u64 uevent_seqnum;
-char uevent_helper[UEVENT_HELPER_PATH_LEN] = "/sbin/hotplug";
-
 #define KERNEL_ATTR_RO(_name) \
 static struct subsys_attribute _name##_attr = __ATTR_RO(_name)
 
--- gregkh-2.6.orig/lib/kobject_uevent.c
+++ gregkh-2.6/lib/kobject_uevent.c
@@ -26,6 +26,8 @@
 #define NUM_ENVP	32	/* number of env pointers */
 
 #if defined(CONFIG_HOTPLUG) && defined(CONFIG_NET)
+u64 uevent_seqnum;
+char uevent_helper[UEVENT_HELPER_PATH_LEN] = "/sbin/hotplug";
 static DEFINE_SPINLOCK(sequence_lock);
 static struct sock *uevent_sock;
 
